cvat/apps/dataset_manager/bindings.py

```diff
@@ -98,10 +98,16 @@ def _get_immutable_attribute_id(self, label_id, attribute_name):
         return self._get_attribute_id(label_id, attribute_name, 'immutable')
 
     def abs_frame_id(self, relative_id):
+        if relative_id not in range(0, self._db_task.data.size):
+            raise ValueError("Unknown internal frame id %s" % relative_id)
         return relative_id * self._frame_step + self._db_task.data.start_frame
 
     def rel_frame_id(self, absolute_id):
-        return (absolute_id - self._db_task.data.start_frame) // self._frame_step
+        d, m = divmod(
+            absolute_id - self._db_task.data.start_frame, self._frame_step)
+        if m or d not in range(0, self._db_task.data.size):
```

Comment:
Can `m or not(0 <= d < self._db_task.data.size)` condition be used here?

Reply:
It can, but I'm not a big fan of it. We also could write `abs_id in range(start_frame, stop_frame, frame_step)`.